More clearly define the default abi of an extern block

ehuss opened the pull request:
This attempts to make it clearer that if the ABI string is not specified, it defaults to the "C" ABI. I felt like the existing text didn't emphasize that enough.

I also took the opportunity to reword the intro to more generally explain what the ABI string does.

A followup PR will further clarify some of the confusing use of the word "default" in the specific ABI definitions.

This is implemented in [`ExternAbi::FALLBACK`](https://github.com/rust-lang/rust/blob/a9d0a6f15533a364816c4d81e2192009ef601d33/compiler/rustc_abi/src/extern_abi.rs#L287)

ehuss committed:
More clearly define the default abi of an extern block

This attempts to make it clearer that if the ABI string is not
specified, it defaults to the "C" ABI. I felt like the existing text
didn't emphasize that enough.

I also took the opportunity to reword the intro to more generally
explain what the ABI string does.

A followup PR will further clarify some of the confusing use of the word
"default" in the specific ABI definitions.

traviscross committed:
Revise wording on explicit extern ABI change

tshepang commented on src/items/external-blocks.md:
````diff
@@ -98,14 +98,19 @@ r[items.extern.abi]
+> unsafe extern "system" { }
+> ```
+
+r[iterms.extern.abi.default]
````
```suggestion
r[items.extern.abi.default]
```

traviscross committed:
Add glossary entry for ABI and use it

Rather than using an `abbr` tag when mentioning ABI, let's add a
glossary entry for ABI and link to that.
